op-acceptance-tests/tests/interop/interop_tx_test.go

```diff
@@ -1,50 +1,132 @@
+		logger = logger.With("test", "TestMessagePassing", "devnet", sys.Identifier())
 
 		chainA := sys.L2s()[sourceChainIdx]
-		chainB := sys.L2s()[destChainIdx]
+		chainB := llsys.L2s()[destChainIdx]
 
-		logger = logger.With("sourceChain", chainA.ID(), "destChain", chainB.ID())
+		logger.Info("chain info", "sourceChain", chainA.ID(), "destChain", chainB.ID())
 
-		// userA is funded at chainA and want to send message to chainB
-		userA := walletGetter(ctx)
+		// userA is funded at chainA and want to initialize message at chain A
+		userA := sourceWalletGetter(ctx)
+		// userB is funded at chainB and want to execute message to chainB
+		userB := destWalletGetter(ctx)
+
+		sha256PrecompileAddr := common.BytesToAddress([]byte{0x2})
+		dummyMessage := []byte("l33t message")
 
 		// Initiate message
-		dummyAddress := common.Address{0x13, 0x37}
-		dummyMessage := []byte{0x13, 0x33, 0x33, 0x37}
-		logger.Info("Initiate message", "address", dummyAddress, "message", dummyMessage)
-		require.NoError(t, userA.InitiateMessage(chainB.ID(), dummyAddress, dummyMessage).Send(ctx).Wait())
+		logger.Info("Initiate message", "address", sha256PrecompileAddr, "message", dummyMessage)
+		initResult := userA.InitiateMessage(chainB.ID(), sha256PrecompileAddr, dummyMessage).Send(ctx)
+		require.NoError(t, initResult.Wait())
+
+		initReceipt, ok := initResult.Info().(system.Receipt)
+		require.True(t, ok)
+		logger.Info("Initiate message", "txHash", initReceipt.TxHash().Hex())
+		logs := initReceipt.Logs()
+		// We are directly calling sendMessage, so we expect single log for SentMessage event
+		require.Equal(t, 1, len(logs), "expected single log")
+		log := logs[0]
+
+		// Build sentMessage for message execution
+		blockNumber := initReceipt.BlockNumber()
+		blockA, err := chainA.Node().BlockByNumber(ctx, blockNumber)
+		require.NoError(t, err)
+		blockTimeA := big.NewInt(int64(blockA.Time()))
+		logger.Info("Initiate message was included at", "timestamp", blockTimeA.String())
+
+		sentMessage := []byte{}
+		for _, topic := range log.Topics {
+			sentMessage = append(sentMessage, topic.Bytes()...)
+		}
+		sentMessage = append(sentMessage, log.Data...)
+
+		// Build identifier for message execution
+		logIndex := big.NewInt(int64(log.Index))
+		identifier := bindings.Identifier{
+			Origin:      constants.L2ToL2CrossDomainMessenger,
+			BlockNumber: blockNumber,
+			LogIndex:    logIndex,
+			Timestamp:   blockTimeA,
+			ChainId:     chainA.ID(),
+		}
+
+		// Execute message
+		logger.Info("Execute message", "address", sha256PrecompileAddr, "message", dummyMessage)
+		execResult := userB.ExecuteMessage(identifier, sentMessage).Send(ctx)
+		require.NoError(t, execResult.Wait())
+
+		execReceipt, ok := execResult.Info().(system.Receipt)
+		require.True(t, ok)
+
+		execTxHash := execReceipt.TxHash()
+		logger.Info("Execute message", "txHash", execTxHash.Hex())
+
+		blockB, err := chainB.Node().BlockByNumber(ctx, blockNumber)
```

Comment:
BlockNumber is set at :57 based on the init message transaction block on the source chain and never reset.

Reply:
Thanks for the catch! fixed at https://github.com/ethereum-optimism/optimism/pull/14838/commits/ba566c12aa5a8c06c27b76d96ef849250852b907

Reply:
was it passing before?

Reply:
Yeah it is passing because that blocknumber is only used for logging.